File: backend/app/schemas/document.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict, Field


Severity = Literal["high", "medium", "low"]


class RequirementFinding(BaseModel):
    id: str
    title: str
    description: str
    severity: Severity = "medium"
    category: str = "需求评审"
    adopted: bool = False


class DocumentBase(BaseModel):
    project_id: Optional[int] = None
    name: str = Field(min_length=1, max_length=255)
    parent_id: Optional[int] = None
    title: Optional[str] = Field(default=None, max_length=255)
    content: Optional[str] = None
    created_by: Optional[str] = Field(default=None, max_length=100)
    is_directory: bool = False
    ai_suggest: list[RequirementFinding] = Field(default_factory=list)


class DocumentCreate(DocumentBase):
    pass


class DocumentUpdate(BaseModel):
    project_id: Optional[int] = None
    name: Optional[str] = Field(default=None, min_length=1, max_length=255)
    parent_id: Optional[int] = None
    title: Optional[str] = Field(default=None, max_length=255)
    content: Optional[str] = None
    created_by: Optional[str] = Field(default=None, max_length=100)
    is_directory: Optional[bool] = None
    ai_suggest: Optional[list[RequirementFinding]] = None


class DocumentOut(DocumentBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class RequirementReviewRequest(BaseModel):
    document_id: Optional[int] = None
    title: Optional[str] = None
    content: str = Field(min_length=1)
    extra_prompt: Optional[str] = Field(default=None, max_length=1000)


class RequirementReviewResponse(BaseModel):
    document_id: Optional[int] = None
    title: Optional[str] = None
    findings: list[RequirementFinding]

File: fronted/src/api/document.ts
import { del, get, post, put } from './http'
import type { AgentTask } from './agentTask'
import { getPageItems, type PaginatedData } from './pagination'

export type FindingSeverity = 'high' | 'medium' | 'low'

export interface RequirementFinding {
  id: string
  title: string
  description: string
  severity: FindingSeverity
  category: string
  adopted: boolean
}

export interface RequirementDocument {
  id: number
  project_id?: number | null
  name: string
  parent_id?: number | null
  title?: string | null
  content?: string | null
  created_by?: string | null
  is_directory: boolean
  ai_suggest: RequirementFinding[]
  created_at?: string | null
  updated_at?: string | null
}

export interface DocumentPayload {
  project_id?: number | null
  name: string
  parent_id?: number | null
  title?: string | null
  content?: string | null
  created_by?: string | null
  is_directory?: boolean
  ai_suggest?: RequirementFinding[]
}

export interface RequirementReviewRequest {
  document_id?: number
  title?: string | null
  content: string
  extra_prompt?: string
}

export interface RequirementReviewResponse {
  document_id?: number | null
  title?: string | null
  findings: RequirementFinding[]
}

export function getDocuments(projectId?: number): Promise<RequirementDocument[]> {
  return get<PaginatedData<RequirementDocument>>(
    '/api/v1/documents/',
    projectId ? { project_id: projectId } : undefined,
  ).then(getPageItems)
}

export function createDocument(data: DocumentPayload): Promise<RequirementDocument> {
  return post<RequirementDocument>('/api/v1/documents/', data)
}

export function updateDocument(
  documentId: number,
  data: Partial<DocumentPayload>,
): Promise<RequirementDocument> {
  return put<RequirementDocument>(`/api/v1/documents/${documentId}`, data)
}

export function deleteDocument(documentId: number): Promise<RequirementDocument> {
  return del<RequirementDocument>(`/api/v1/documents/${documentId}`)
}

export function reviewRequirement(
  data: RequirementReviewRequest,
): Promise<AgentTask<RequirementReviewResponse>> {
  return post<AgentTask<RequirementReviewResponse>>('/api/v1/documents/review', data)
}
